R/lump.R

```diff
@@ -38,17 +40,33 @@
 #' fct_lump(x, n = -3)
 #' fct_lump(x, prop = -0.1)
 #'
+#' # Use weighted frequencies
+#' w <- c(rep(2, 50), rep(1, 50))
+#' fct_lump(x, n = 5, weights = w)
+#'
 #' # Use ties.method to control how tied factors are collapsed
 #' fct_lump(x, n = 6)
 #' fct_lump(x, n = 6, ties.method = "max")
 #'
-fct_lump <- function(f, n, prop, other_level = "Other",
+fct_lump <- function(f, n, prop, weights, other_level = "Other",
                      ties.method = c("min", "average", "first", "last", "random", "max")) {
   f <- check_factor(f)
   ties.method <- match.arg(ties.method)
 
+  if (! missing(weights)) {
+    if (! is.numeric(weights)) {
+      stop("`weights` must be a numeric vector")
+    } else if (! length(f) == length(weights)) {
+      stop("different lengths of `f` and `weights`")
```

Comment:
Error messages should start with a capital letter and always use `call. = FALSE`